src/libraries/Microsoft.Extensions.Logging.Configuration/ref/Microsoft.Extensions.Logging.Configuration.cs

```diff
@@ -23,7 +23,8 @@ public partial interface ILoggerProviderConfiguration<T>
     }
     public static partial class LoggerProviderOptions
     {
-        public static void RegisterProviderOptions<TOptions, TProvider>(Microsoft.Extensions.DependencyInjection.IServiceCollection services) where TOptions : class { }
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode("The type of TOptions cannot be statically analyzed so its members may be trimmed.")]
```

Comment:
I think the message is somewhat misleading. Generic parameters are typically statically analyzable, that's not the problem here. The problem is that we don't have annotation which would recursively preserve enough things on TOptions to work all the time.

Maybe something more generic:
"Trimming may remove some parts of the TOptions and it's dependent types"

Can't come up with something good right now, hopefully somebody will have a good idea.

Reply:
I'd also add a suggestion of what to do in order to make it trim-compatible. Something like "Ensure the necessary constructors and properties are preserved"